l10n_ch_hr_payroll/models/hr_contract.py

```diff
@@ -27,37 +27,107 @@
 ##############################################################################
 
 
-from openerp.osv import fields, orm
+from openerp import models, fields, api
 import openerp.addons.decimal_precision as dp
 
 import logging
 _logger = logging.getLogger(__name__)
 
 
-class hr_contract(orm.Model):
+class HrContract(models.Model):
     _inherit = 'hr.contract'
-    _columns = {
-        'lpp_rate': fields.float('LPP Rate',
-                                 digits_compute=dp.get_precision(
-                                     'Payroll Rate')),
-        'lpp_amount': fields.float('LPP Amount',
-                                   digits_compute=dp.get_precision(
-                                       'Account')),
-        'worked_hours': fields.float('Worked Hours'),
-        'hourly_rate': fields.float('Hourly Rate'),
-        'holiday_rate': fields.float('Holiday Rate')
-        }
 
-    def compute_wage(self, cr, uid, id,
-                     worked_hours, hourly_rate,
-                     context=None):
+    # ---------- Fields management
+
+    @api.one
```

Comment:
I have added the user linked to the employee. 



Avec mes meilleures salutations Yvon-Philippe Crittin Senior Developper, OpenERP Specialist Open Net Sàrl www.open-net.ch Accélérateur de productivité 59, Rue de l'industrie CH-1030 Bussigny Autoroute A1, sortie Bussigny Bus TL 17, arrêt Croix-de-Plan T: +41 (21) 701 42 45 
----- Mail original -----

De: "Yannick Vaucher" <notifications@github.com> 
À: "OCA/l10n-switzerland" <l10n-switzerland@noreply.github.com> 
Cc: "Open Net Ltd" <info@open-net.ch> 
Envoyé: Mardi 6 Octobre 2015 18:49:29 
Objet: Re: [l10n-switzerland] New: commissions and reimbursements mgt in payslips and contracts (#168) 


In l10n_ch_hr_payroll/models/hr_contract.py : >  
> -    def compute_wage(self, cr, uid, id,
> -                     worked_hours, hourly_rate,
> -                     context=None):
> +    # ---------- Fields management
> +
> +    @api.one 
Can you list the fields on which this computation depends with @api.depends 
Or is there any reason not to do it? Is it because it isn't stored? 
https://www.odoo.com/documentation/8.0/reference/orm.html#computed-fields 
If we take this non stored field as an example api.depends is defined: 
https://github.com/odoo/odoo/blob/8.0/addons/account/account_invoice.py#L295-L296 
https://github.com/odoo/odoo/blob/8.0/addons/account/account_invoice.py#L149-L155 
Not sure why thought, maybe to update the cache. 
— 
Reply to this email directly or view it on GitHub .